src/ctapipe/io/hdf5monitoringsource.py

```diff
@@ -187,38 +182,27 @@ def __init__(
             PointingInterpolator,
         )
 
-        # Set the subarray description
-        self.subarray = subarray
-
-        # Handle additional input_files parameter - extend the existing list
-        if input_files is not None:
-            if isinstance(input_files, (list, tuple)):
-                self.input_files.extend([pathlib.Path(f) for f in input_files])
-            else:
-                self.input_files.append(pathlib.Path(input_files))
         # Check if input_files list is empty
         if not self.input_files:
-            raise IOError(
-                "No input files provided. Please specify input files "
-                "via configuration or input_files parameter."
+            raise InputMissing(
+                "No input files provided. Please specify a list of input file(s) "
+                "via configuration by `--HDF5MonitoringSource.input_files` "
+                "or using as an argument <input_files> in the constructor."
             )
         # Loop over the input files to read the subarray description and check for compatibility
         # if a subarray is already provided either externally or via a previous monitoring file.
+        subarray_list = [] if self.subarray is None else [self.subarray]
         for file in self.input_files:
             # Log the input file paths
             self.log.info("INPUT PATH = %s", str(file))
             # Read the subarray description from the monitoring file
-            subarray_from_file = SubarrayDescription.from_hdf(file)
-            if self.subarray is not None and any(
-                self.subarray.tel_ids != subarray_from_file.tel_ids
-            ):
-                raise ToolConfigurationError(
-                    f"HDF5MonitoringSource: Available telescopes '{subarray_from_file.tel_ids}' "
-                    f"from monitoring file '{file}' are not compatible with: '{self.subarray.tel_ids}'."
-                )
-
-            # Overwrite or set the subarray description
-            self.subarray = subarray_from_file
+            subarray_list.append(SubarrayDescription.from_hdf(file))
+        # Check if all subarray descriptions are compatible
+        if not SubarrayDescription.check_matching_subarrays(subarray_list):
+            raise IOError("Incompatible subarray descriptions found in input files.")
+        else:
```

Comment:
You don't need this "else" statement, as you raise on the condition above. Remove it and deindent the next line.